frontend/src/lib/search.ts

```diff
@@ -72,6 +72,25 @@ export function queryRemoteArgoApps(): IRequestResult<ISearchResult> {
   })
 }
 
+export function queryRemoteArgoAppsForCluster(cluster: string): IRequestResult<ISearchResult> {
+  return postRequest<SearchQuery, ISearchResult>(getBackendUrl() + apiSearchUrl, {
+    operationName: 'searchResult',
+    variables: {
+      input: [
+        {
+          filters: [
+            { property: 'kind', values: ['Application'] },
+            { property: 'apigroup', values: ['argoproj.io'] },
+            { property: 'cluster', values: [cluster] },
+          ],
+          limit: 20000,
+        },
+      ],
+    },
+    query: searchFilterQuery,
+  })
+}
+
```

Comment:
@fxiang1 Overall the PR looks great. I think it would be cleaner to combine this new function with the original, making the `cluster` param optional, both in terms of reducing duplication and simplifying where it is used in the code. Same thing for `queryOCPAppResourcesForCluster`.

Reply:
Thanks for catching this @KevinFCormier, I will work on it.